### PR TITLE
Fix leaking goroutines in multiple integration tests

### PR DESCRIPTION
**Pull request by wojtek-t**
Ref https://github.com/kubernetes/kubernetes/issues/108483

Note to reviewer: I recommend reviewing commits one by one - they are all independent.

```release-note
NONE
```

/kind cleanup
/priority important-longterm
/sig testing

### CODE REVIEW

**Comment by aojea** [comment]
```


1/43 Tests Failed. | expand_less
-- | --
verify: gofmt expand_less6s{Script Error ScriptError diff -u ./test/integration/network/services_test.go.orig ./test/integration/network/services_test.go --- ./test/integration/network/services_test.go.orig	2022-06-02 19:27:57.762267425 +0000 +++ ./test/integration/network/services_test.go	2022-06-02 19:27:57.762267425 +0000 @@ -48,7 +48,6 @@  		ModifyServerConfig: func(cfg *controlplane.Config) {  			cfg.ExtraConfig.RepairServicesInterval = interval  		}, -  	})  	defer tearDownFn()    Run ./hack/update-gofmt.sh} | verify: gofmt expand_less | 6s | {Script Error ScriptError diff -u ./test/integration/network/services_test.go.orig ./test/integration/network/services_test.go --- ./test/integration/network/services_test.go.orig	2022-06-02 19:27:57.762267425 +0000 +++ ./test/integration/network/services_test.go	2022-06-02 19:27:57.762267425 +0000 @@ -48,7 +48,6 @@  		ModifyServerConfig: func(cfg *controlplane.Config) {  			cfg.ExtraConfig.RepairServicesInterval = interval  		}, -  	})  	defer tearDownFn()    Run ./hack/update-gofmt.sh}
verify: gofmt expand_less | 6s
{Script Error ScriptError diff -u ./test/integration/network/services_test.go.orig ./test/integration/network/services_test.go --- ./test/integration/network/services_test.go.orig	2022-06-02 19:27:57.762267425 +0000 +++ ./test/integration/network/services_test.go	2022-06-02 19:27:57.762267425 +0000 @@ -48,7 +48,6 @@  		ModifyServerConfig: func(cfg *controlplane.Config) {  			cfg.ExtraConfig.RepairServicesInterval = interval  		}, -  	})  	defer tearDownFn()    Run ./hack/update-gofmt.sh}


```

File: pkg/controller/disruption/disruption.go [hunk 1]
```diff
@@ -358,7 +358,18 @@ func verifyGroupKind(controllerRef *metav1.OwnerReference, expectedKind string,
...
 	defer utilruntime.HandleCrash()
+
+	// Start events processing pipeline.
+	if dc.kubeClient != nil {
+		klog.Infof("Sending events to api server.")
+		dc.broadcaster.StartRecordingToSink(&v1core.EventSinkImpl{Interface: dc.kubeClient.CoreV1().Events("")})
+	} else {
+		klog.Infof("No api server defined - no events will be sent to API server.")
+	}
+	defer dc.broadcaster.Shutdown()
```
Comment (aojea) [inline pkg/controller/disruption/disruption.go:369]: do you defer `dc.broadcaster.Shutdown()` always or only if it "startedRecordingToSink" ?

File: test/integration/disruption/disruption_test.go [hunk 2]
```diff
@@ -498,9 +502,15 @@ func waitToObservePods(t *testing.T, podInformer cache.SharedIndexInformer, podN
 }
 
 func TestPatchCompatibility(t *testing.T) {
-	s, _, _, clientSet, _, _ := setup(t)
+	s, pdbc, _, clientSet, _, _ := setup(t)
```
Comment (aojea) [inline test/integration/disruption/disruption_test.go:505]: don't you need to start the informers too?

File: pkg/controller/certificates/certificate_controller.go [hunk 3]
```diff
@@ -60,11 +58,6 @@ func NewCertificateController(
 	csrInformer certificatesinformers.CertificateSigningRequestInformer,
 	handler func(context.Context, *certificates.CertificateSigningRequest) error,
 ) *CertificateController {
-	// Send events to the apiserver
-	eventBroadcaster := record.NewBroadcaster()
-	eventBroadcaster.StartStructuredLogging(0)
-	eventBroadcaster.StartRecordingToSink(&v1core.EventSinkImpl{Interface: kubeClient.CoreV1().Events("")})
```
Comment (aojea) [inline pkg/controller/certificates/certificate_controller.go:66]: ?

File: test/integration/certificates/duration_test.go [hunk 4]
```diff
@@ -51,12 +51,12 @@ import (
...
 	t.Parallel()
 
-	ctx, cancel := context.WithTimeout(context.Background(), 3*time.Minute)
-	t.Cleanup(cancel)
-
 	s := kubeapiservertesting.StartTestServerOrDie(t, nil, nil, framework.SharedEtcd())
 	t.Cleanup(s.TearDownFn)
 
+	ctx, cancel := context.WithTimeout(context.Background(), 3*time.Minute)
+	t.Cleanup(cancel)
```
Comment (aojea) [inline test/integration/certificates/duration_test.go:58]: so many t.Cleanup() functions is not complex to follow?

**Comment by dims** [comment]
looks like need to `Run ./hack/update-gofmt.sh` @wojtek-t

File: pkg/controller/certificates/certificate_controller.go (hunk 3 above)
Comment (wojtek-t) [reply pkg/controller/certificates/certificate_controller.go:66]: This just wasn't used anywhere :D

File: pkg/controller/disruption/disruption.go (hunk 1 above)
Comment (wojtek-t) [reply pkg/controller/disruption/disruption.go:369]: we defer it always (it's unrelated to whether we record to sink or not)

File: test/integration/certificates/duration_test.go (hunk 4 above)
Comment (wojtek-t) [reply test/integration/certificates/duration_test.go:58]: somewhat - I was just trying to do minimal changes...

File: test/integration/disruption/disruption_test.go (hunk 2 above)
Comment (wojtek-t) [reply test/integration/disruption/disruption_test.go:505]: we don't - just creating informers doesn't start any goroutines - so we don't need to touch them

**Review by wojtek-t — COMMENTED** [review]
@aojea - thanks for comments - PTAL

**Pull request merged**